### ISSUE TITLE
Shorty-Id - Fix in BinaryExporterServlet to support Legacy Assets in order to Prevent Images Missing on Page

### ISSUE THREAD

**Issue by agomez-dotcms** [dotCMS/core#10989]
After a [dotCMS upgrade for a specific Client](https://my.dotcms.com/tickets/detail.dot?id=196f6f8a-e468-4f2d-83ae-d283ee4fb6e6), an issue with images missing on a specific page arose. This problem was fixed via hotfix mechanism but it may be fixed at dotCMS core level: **dotCMS core's BinaryExporterServlet is not serving images from legacy-assets when they miss their shorty-id** (for instance, rendered on page like "/dotAsset/2078967.jpg", "/dotAsset/2079125.jpg", "/dotAsset/2079121.jpg").

## Expected Behavior
BinaryExporterServlet (on behalf of SpeedyAssetServlet) should serve legacy assets that have valid data in database (ie. in identifier, file_asset and fileasset_version_info tables)

## Current Behavior
BinaryExporterServlet (on behalf of SpeedyAssetServlet) is returning a 404 HTTP error for each of the images referenced in page

## Possible Solution
Following are the errors displayed in dotcms.log, which suggested that SpeedyAssetServlet and BinaryExporterServlet needed to be code-reviewed in order to understand the causes of the messages:
```
[09/03/17 12:08:29:309 EST]  WARN shorty.ShortyIdAPIImpl: shorty 2101501 is not a short id.  Short Ids should be 10 chars in length
[09/03/17 12:08:29:311 EST] ERROR servlets.BinaryExporterServlet: Error loading object from Hibernate Session

[09/03/17 12:08:29:318 EST]  WARN shorty.ShortyIdAPIImpl: shorty 2078965 is not a short id.  Short Ids should be 10 chars in length
[09/03/17 12:08:29:319 EST] ERROR servlets.BinaryExporterServlet: Error loading object from Hibernate Session 

[09/03/17 12:08:29:327 EST]  WARN shorty.ShortyIdAPIImpl: shorty 2079165 is not a short id.  Short Ids should be 10 chars in length
[09/03/17 12:08:29:327 EST] ERROR servlets.BinaryExporterServlet: Error loading object from Hibernate Session 

[09/03/17 12:08:29:336 EST]  WARN shorty.ShortyIdAPIImpl: shorty 2079119 is not a short id.  Short Ids should be 10 chars in length
[09/03/17 12:08:29:336 EST] ERROR servlets.BinaryExporterServlet: Error loading object from Hibernate Session 

[09/03/17 12:08:29:344 EST]  WARN shorty.ShortyIdAPIImpl: shorty 2078999 is not a short id.  Short Ids should be 10 chars in length
[09/03/17 12:08:29:345 EST] ERROR servlets.BinaryExporterServlet: Error loading object from Hibernate Session 

[09/03/17 12:08:29:360 EST]  WARN shorty.ShortyIdAPIImpl: shorty 2079123 is not a short id.  Short Ids should be 10 chars in length
[09/03/17 12:08:29:361 EST] ERROR servlets.BinaryExporterServlet: Error loading object from Hibernate Session 

[09/03/17 12:08:30:200 EST]  WARN shorty.ShortyIdAPIImpl: shorty 2079169 is not a short id.  Short Ids should be 10 chars in length
[09/03/17 12:08:30:201 EST] ERROR servlets.BinaryExporterServlet: Error loading object from Hibernate Session 

[09/03/17 12:08:30:209 EST]  WARN shorty.ShortyIdAPIImpl: shorty 2078965 is not a short id.  Short Ids should be 10 chars in length
[09/03/17 12:08:30:210 EST] ERROR servlets.BinaryExporterServlet: Error loading object from Hibernate Session 

[09/03/17 12:08:30:217 EST]  WARN shorty.ShortyIdAPIImpl: shorty 2079123 is not a short id.  Short Ids should be 10 chars in length
[09/03/17 12:08:30:218 EST] ERROR servlets.BinaryExporterServlet: Error loading object from Hibernate Session 

[09/03/17 12:08:30:226 EST]  WARN shorty.ShortyIdAPIImpl: shorty 2079165 is not a short id.  Short Ids should be 10 chars in length
[09/03/17 12:08:30:227 EST] ERROR servlets.BinaryExporterServlet: Error loading object from Hibernate Session 

[09/03/17 12:08:30:235 EST]  WARN shorty.ShortyIdAPIImpl: shorty 2079119 is not a short id.  Short Ids should be 10 chars in length
[09/03/17 12:08:30:236 EST] ERROR servlets.BinaryExporterServlet: Error loading object from Hibernate Session 

[09/03/17 12:08:30:244 EST]  WARN shorty.ShortyIdAPIImpl: shorty 2078999 is not a short id.  Short Ids should be 10 chars in length
[09/03/17 12:08:30:245 EST] ERROR servlets.BinaryExporterServlet: Error loading object from Hibernate Session 
```

Following was the fix made, which may be incorporated as part of dotCMS's core (Added "**uuid = (!ShortType.CACHE_MISS.toString().equals(shorty.longId)) ? shorty.longId : shorty.shortId;**" instead of "**uuid = shorty.longId;**"):
https://github.com/dotCMS/core/blob/3.7.1/src/com/dotmarketing/servlets/BinaryExporterServlet.java#L165

```
public class BinaryExporterServlet extends HttpServlet {
   ...
   @Override
   public void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
      ...
      String uuid = uriPieces[2];
      ...
      boolean isContent= (shorty.subType == ShortType.CONTENTLET);
      uuid = (!ShortType.CACHE_MISS.toString().equals(shorty.longId)) ? shorty.longId : shorty.shortId;
      ...
   }
   ...
}
```

## Context
Reported by client after upgrade to dotCMS 3.7.1. https://my.dotcms.com/tickets/detail.dot?id=196f6f8a-e468-4f2d-83ae-d283ee4fb6e6

## Your Environment
* dotCMS version used: 3.7.1.
* Browser Name and version: Any Browser
* Operating System and version: Amazon Linux
* Application Server and version: Apache Tomcat
* Database and version: Postgres

**Comment by agomez-dotcms** [comment]
The fix for this issue may be borrowed from hot-fix at https://github.com/dotCMS/support/tree/master/hotfixes/hotfix-GIT-10989_v3.7.1. Also, the original Support ticket may be accessed at https://my.dotcms.com/tickets/detail.dot?id=196f6f8a-e468-4f2d-83ae-d283ee4fb6e6

**Comment by jgambarios** [comment]
Closed as we removed in 4.0 the support for legacy files. 
Related to: https://github.com/dotCMS/core/issues/10716
